### PR TITLE
feat/fix: Button component with tests and a few minor fixes

### PR DESCRIPTION
**Pull request by j24m**
## Issue : 
- https://github.com/Real-Dev-Squad/tiny-site-frontend/issues/20

## Description :
- Added a new React component called Button.
- Defined a ButtonProps interface to specify the component's props, including type, className, onClick, and children.
- The Button component renders a button element with the specified props.

## Testing : 
- The Button component has been tested using the React Testing Library and Jest. Two test cases have been added.
- **Before :**
image

 
- **After :**
image

## Working Proof :
- **Dashboard :** 
image

- **Login :** 
image

### CODE REVIEW

File: src/pages/dashboard/index.js
Comment (iamitprakash) [inline src/pages/dashboard/index.js:29]: please remove this commented code

File: src/pages/dashboard/index.js
Comment (iamitprakash) [inline src/pages/dashboard/index.js:65]: same here

File: src/pages/dashboard/index.js
Comment (iamitprakash) [inline src/pages/dashboard/index.js:91]: same here

File: __tests__/components/Button.test.tsx
Comment (kotesh-arya) [inline __tests__/components/Button.test.tsx:18]: This snapshot testing generates an instance of the rendered component, each time you run the test and we don't need these snapshots while pushing the code. The better alternative will be 
```
   const { getByRole } = render(
            <Button type="submit"/>
        );

   const buttonElement = getByRole('button');
   expect(buttonElement).toHaveAttribute('type', 'submit');
```

File: __tests__/components/Button.test.tsx
Comment (kotesh-arya) [inline __tests__/components/Button.test.tsx:29]: As you're passing the `children` prop, you don't need the closing tag for the Button component

File: src/components/Button/index.tsx
Comment (sahsisunny) [inline src/components/Button/index.tsx:1]: Can you set the default style of the button? And also set the default type of the button "button".

File: src/pages/login/index.js
Comment (sahsisunny) [inline src/pages/login/index.js:88]:
```suggestion
                            Log in
```

File: public/assets/icons/add.tsx
Comment (Pratiyushkumar) [inline public/assets/icons/add.tsx:4]: why is this change required....any specific reason?
I think previous code also looked good

File: public/assets/icons/copy.tsx
Comment (Pratiyushkumar) [inline public/assets/icons/copy.tsx:7]: similarly here as well

File: src/components/Button/index.tsx
Comment (Pratiyushkumar) [reply src/components/Button/index.tsx:1]: can you also put interfaces into separate folder since we are refactoring the code base.

File: __tests__/components/Button.test.tsx
Comment (Pratiyushkumar) [inline __tests__/components/Button.test.tsx:7]: please correct the `it` statement, "should render" what??

**Review by Pratiyushkumar — COMMENTED** [review]
Also, try to verify your changes before pushing your code into PR this will avoid unnecessary changes to your PR. Example - NavBar.tsx
this will also avoid increase in number of lines changed

File: src/components/Button/index.tsx
Comment (vinit717) [inline src/components/Button/index.tsx:7]:
```suggestion
    children: React.ReactNode;
```

File: src/components/Button/index.tsx
Comment (shubhamsigdar1) [inline src/components/Button/index.tsx:10]: Nit: I think you have to pass disabled prop in Button component
correct me if i am wrong

**Review by shubhamsigdar1 — COMMENTED** [review]
Thank you for great description
and great work

**Comment by gauravsinhaweb** [comment]
Don't merge this PR before merging this
- https://github.com/Real-Dev-Squad/tiny-site-frontend/pull/18

File: __tests__/components/Button.test.tsx
Comment (j24m) [reply __tests__/components/Button.test.tsx:29]: Done.

File: src/components/Button/index.tsx
Comment (sahsisunny) [reply src/components/Button/index.tsx:1]: > can you also put interfaces into separate folder since we are refactoring the code base.

No need to write the separate the types or interfaces for now.

File: public/assets/icons/add.tsx
Comment (j24m) [reply public/assets/icons/add.tsx:4]: This is a syntax convention.
image

File: public/assets/icons/copy.tsx
Comment (j24m) [reply public/assets/icons/copy.tsx:7]: Replied above.

File: __tests__/components/Button.test.tsx
Comment (j24m) [reply __tests__/components/Button.test.tsx:7]: done.

File: src/components/Button/index.tsx
Comment (j24m) [reply src/components/Button/index.tsx:7]: Done.

File: src/components/Button/index.tsx
Comment (j24m) [reply src/components/Button/index.tsx:10]: Depends on the use case. Here there is no need to disable it as of now.

File: __tests__/components/Button.test.tsx
Comment (j24m) [reply __tests__/components/Button.test.tsx:18]: Done.

File: src/components/Button/index.tsx
Comment (j24m) [reply src/components/Button/index.tsx:1]: > Can you set the default style of the button? And also set the default type of the button "button".

- I am curious as to why we need to set a default style for buttons, mostly we will always pass styles while using a component, so default styles will always be overridden.
- Even if I do pass a default value for style and let's assume that a developer forgets to provide styling while using the component, at different places the requirement for styling is different. No button has common styling.
- So what according to you should be passed as a default value ?

File: src/pages/login/index.js
Comment (j24m) [reply src/pages/login/index.js:88]: Done.

File: src/components/Button/index.tsx
Comment (shubhamsigdar1) [reply src/components/Button/index.tsx:10]: You made reusable component...Right??

**Review by sahsisunny — DISMISSED** [review]

**Review by vinit717 — APPROVED** [review]

**Review by sahsisunny — APPROVED** [review]

**Review by shubhamsigdar1 — APPROVED** [review]

**Review by vinit717 — APPROVED** [review]

**Pull request merged**
